addons/website_sale_slides/views/website_slides_templates.xml

```diff
@@ -79,4 +79,39 @@
         </div>
     </xpath>
 </template>
+
+<template name="Buy Course To Download Resource" id="slide_aside_training_category_buy_course" inherit_id="website_slides.slide_aside_training_category">
+    <xpath expr="//div[hasclass('o_wslides_js_course_join')]" position="inside">
+        <li t-elif="not is_member and aside_slide.channel_id.enroll == 'payment'" class="text-decoration-none small">
+            <i class="fa fa-download mr-1"/>
+            <a t-att-href="'/shop/cart/update?product_id=%s' % aside_slide.channel_id.product_id.id">
+                Buy Course
+            </a>
+             to download resource
+        </li>
+    </xpath>
+</template>
+
+<template name="Buy Course To Download Resource Fullscreen" id="slide_fullscreen_sidebar_category_buy_course" inherit_id="website_slides.slide_fullscreen_sidebar_category">
+    <xpath expr="//div[hasclass('o_wslides_js_course_join')]" position="inside">
+        <li t-elif="not is_member and slide.channel_id.enroll == 'payment'" class="o_wslides_fs_slide_link mb-1">
+            <i class="fa fa-download mr-1"/>
+            <a t-att-href="'/shop/cart/update?product_id=%s' % slide.channel_id.product_id.id">
+                Buy Course
+            </a>
+             to download resource
+        </li>
```

Comment:
I created a template for the link but not the rest since the icon and the li is used for only 2 cases the third case uses a span and no icon. Does it seems good to you ?